### PR TITLE
Update prometheus and promql-engine dependencies

### PR DESCRIPTION
**Pull request by harry671003**
* [ ] I added CHANGELOG entry for this change.
* [X] Change is not relevant to the end user.

## Changes
- Updating prometheus to e9dec5fc537b
- Updating thanos-io/promql-engine to cdbd6dfed36b

## Verification

- Build was successful

## Depends on
- https://github.com/thanos-io/thanos/pull/7559

### CODE REVIEW

**Review by saswatamcode — DISMISSED** [review]
LGTM but I think there were some unresolved comments in previous bump PR https://github.com/thanos-io/thanos/pull/7517#discussion_r1674376290 cc: @fpetkovski

File: internal/cortex/querier/queryrange/split_by_interval_test.go [hunk 1]
```diff
@@ -373,11 +374,6 @@ func Test_evaluateAtModifier(t *testing.T) {
 				[2m:])
 			[10m:])`,
 		},
-		{
-			// parse error: missing unit character in duration
```
Comment (harry671003) [inline internal/cortex/querier/queryrange/split_by_interval_test.go:377]: See: https://github.com/prometheus/prometheus/pull/9138

**Review by fpetkovski — APPROVED** [review]
This picks up a commit which has the correct compactor method signature, so we should be good to merge.

**Review by yeya24 — APPROVED** [review]
Thanks

**Pull request merged**